### PR TITLE
user-defined prefixes for label aggregation

### PR DESCRIPTION
**Pull request by yinsenyan**
#### What type of PR is this?
kind/feature

#### What this PR does / why we need it:
In old version , we hard code label prefix for aggregate.
We need aggregate label by custom in more cases.

#### Which issue(s) this PR fixes:
If there are multiple key-value pairs with same key whose occurrence ratio is greater than the threshold, we need to aggregate the key-value pairs that appear more times, not the last one.

### CODE REVIEW

**Review by dixudx — COMMENTED** [review]
Good work!

There is room for improvement in certain sections of the code.

**Comment by dixudx** [comment]
cc @lmxia for a review

File: pkg/controllers/clusters/clusterstatus/clusterstatus_controller_test.go [hunk 1]
```diff
@@ -127,3 +130,127 @@ func (suite *StatusSuite) TestAggregateLimitedSingleLabels() {
...
+			},
+			want: map[string]string{"foo": "bar", "zone": "zone0"},
+		},
+		{
+			name: "aggretate with mutil prefix case 2",
+			args: arg{
+				nodes: []*corev1.Node{
+					{ObjectMeta: metav1.ObjectMeta{
+						Name:   "n0",
+						Labels: map[string]string{"foo": "bar", "zone": "zone0"},
```
[the remark this answers is not in the document]
Comment (yinsenyan) [reply pkg/controllers/clusters/clusterstatus/clusterstatus_controller_test.go:233]: Add one case for foo/abc

File: pkg/agent/status_manager.go [hunk 2]
```diff
@@ -104,6 +107,7 @@ func (mgr *Manager) Run(ctx context.Context, parentDedicatedKubeConfig *rest.Con
 }
 
 func (mgr *Manager) updateClusterStatus(ctx context.Context, namespace, clusterID string, client clusternetclientset.Interface, backoff wait.Backoff) {
+	klog.V(4).Info("will update cluster status: ", mgr.managedCluster)
```
[the remark this answers is not in the document]
Comment (yinsenyan) [reply pkg/agent/status_manager.go]: Update to show namespace and name

File: pkg/agent/options/cluster_reg_options.go [hunk 3]
```diff
@@ -109,6 +114,8 @@ func (opts *ClusterRegistrationOptions) AddFlagSets(fss *cliflag.NamedFlagSets)
 		"Specify the labels for the child cluster, split by `,`")
 	mgmtfs.Float32Var(&opts.LabelAggregateThreshold, LabelAggregateThreshold, opts.LabelAggregateThreshold,
 		"Specifies the threshold of common labels in child cluster nodes should be aggregated to parent")
+	mgmtfs.StringArrayVar(&opts.LabelAggregatePrefix, LabelAggregatePrefix, opts.LabelAggregatePrefix,
+		"Specifies some label prefix who will update to parent cluster")
```
Comment (dixudx) [inline pkg/agent/options/cluster_reg_options.go:118]: Specifies the desired prefixes of node labels to be aggregated

File: pkg/controllers/clusters/clusterstatus/clusterstatus_controller.go [hunk 4]
```diff
@@ -388,37 +389,58 @@ func isMasterNode(labels map[string]string) bool {
...
-					countMap[countKey] = total + 1
+			if utils.ContainsPrefix(labelPrefixes, k) {
+				label := strings.Join([]string{k, v}, "=")
+				if labelMap, isok := countMap[k]; isok {
+					if count, isok2 := labelMap[label]; isok2 {
+						labelMap[label] = count + 1
+					} else {
+						labelMap[label] = 1
+					}
+					countMap[k] = labelMap
```
Comment (dixudx) [inline pkg/controllers/clusters/clusterstatus/clusterstatus_controller.go:410]: This is redundant.

File: pkg/controllers/clusters/clusterstatus/clusterstatus_controller.go [hunk 5]
```diff
@@ -388,37 +389,58 @@ func isMasterNode(labels map[string]string) bool {
...
 	// statistic map.
-	countMap := make(map[string]int, 0)
+	countMap := make(map[string]map[string]int, 0)
 	masterNodeNum := 0
 	for _, node := range nodes {
 		if isMasterNode(node.Labels) {
 			masterNodeNum += 1
 			continue
 		}
 		for k, v := range node.Labels {
```
Comment (dixudx) [inline pkg/controllers/clusters/clusterstatus/clusterstatus_controller.go:401]: Could we rename `k` and `v` to make them more readable?

**Comment by yinsenyan** [comment]
@dixudx How to re-run make test ?

File: pkg/controllers/clusters/clusterstatus/clusterstatus_controller_test.go [hunk 6]
```diff
@@ -127,3 +130,149 @@ func (suite *StatusSuite) TestAggregateLimitedSingleLabels() {
...
+	var tests = []struct {
+		name string
+		args arg
+		want map[string]string
+	}{
+		{
+			name: "aggregate one",
+			args: arg{
+				nodes: []*corev1.Node{
+					{
```
Comment (dixudx) [inline pkg/controllers/clusters/clusterstatus/clusterstatus_controller_test.go:149]: We could use function `BuildNode` to improve the code quality.

**Review by dixudx — APPROVED** [review]
Mostly looks good to me.

File: pkg/controllers/clusters/clusterstatus/clusterstatus_controller_test.go (hunk 6 above)
Comment (yinsenyan) [reply pkg/controllers/clusters/clusterstatus/clusterstatus_controller_test.go]: OK

**Review by dixudx — APPROVED** [review]
Nice work!

**Pull request merged**